components/financial_assistance/app/models/financial_assistance/application.rb

```diff
@@ -1640,14 +1693,27 @@ def verification_update_for_applicants
       end
     end
 
+    # Sets the submission details and calculates necessary values for the application.
+    #
+    # @note This method sets data and builds objects, hitting the database only once instead of multiple times for each method.
+    #
+    # @return [void]
     def set_submit
```

Comment:
Totally not necessary here, but a thought for the future: we could have a similar clean up and optimization done for the `unset_submit`method.

Reply:
Yes. We must optimize the `unset_submit ` method. I am planning to create a new ticket for this.